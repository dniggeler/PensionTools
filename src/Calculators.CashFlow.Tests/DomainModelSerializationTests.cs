using System;
using System.Collections.Generic;
using System.Linq;
using Calculators.CashFlow.Models;
using Newtonsoft.Json;
using PensionCoach.Tools.CommonTypes;
using PensionCoach.Tools.CommonTypes.MultiPeriod;
using PensionCoach.Tools.CommonTypes.Municipality;
using PensionCoach.Tools.CommonTypes.Tax;
using Snapshooter.Xunit;
using Xunit;

namespace Calculators.CashFlow.Tests
{
    [Trait("Domain Model", "Serialization")]
    public class DomainModelSerializationTests
    {
        [Fact(DisplayName = "Deserialize Municipality")]
        public void Deserialize_MunicipalityModel()
        {
            // given
            MunicipalityModel model = new MunicipalityModel
            {
                BfsNumber = 261,
                Canton = Canton.ZH,
                Name = "Zürich",
                MutationId = 1911,
                SuccessorId = 33,
                DateOfMutation = new DateTime(2020,1,1)
            };
            string json = JsonConvert.SerializeObject(new[]{ model });

            // when
            var result = JsonConvert.DeserializeObject<IEnumerable<MunicipalityModel>>(json);

            // then
            Snapshot.Match(result);
        }
    }
}
