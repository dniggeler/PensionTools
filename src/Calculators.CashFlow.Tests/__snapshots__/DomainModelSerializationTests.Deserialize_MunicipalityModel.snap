[
  {
    "BfsNumber": 261,
    "Name": "Zürich",
    "Canton": "ZH",
    "DateOfMutation": "2020-01-01T00:00:00",
    "MutationId": 1911,
    "SuccessorId": 33
  }
]
